src/include/ipc/control.h

```diff
@@ -164,10 +164,12 @@ struct sof_ipc_comp_event {
 	uint32_t src_comp_id;	/**< source component id */
 	uint32_t event_type;	/**< event type - SOF_CTRL_EVENT_* */
 	uint32_t num_elems;	/**< in array elems or bytes for data type */
-
+	/* Callbacks */
+	void (*cb)(void *data);
```

Comment:
There should be no function pointers in IPC, cos other side will not access that function anyway